File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = INT_MIN;
        int currSum = 0;
        for(int x : nums){
            currSum += x;
            if(currSum > maxSum){
                maxSum = currSum;
            }
            if(currSum<0) currSum = 0;
        }
        return maxSum;
    }
};

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum = nums[0]
        currentSum = 0
        
        for i in nums:
            currentSum += i
            maxSum = max(maxSum,currentSum)
            
            if(currentSum<0):
                currentSum = 0
        
        return maxSum